calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+func detectIPs(ignoreInterfaces []string) ([]string, error) {
+
+	ips := []string{}
+	ifaces, err := net.Interfaces()
+
+	for _, iface := range ifaces {
+
+		valid_ip := true
+
+		// Check each interface name against 
+		// list of interfaces to ignore.
+		for _, v := range ignoreInterfaces {
+			regex, err := regexp.Compile(v)
+
+			if err != nil {
+				continue
+			}
+
+			if regex.MatchString(iface.Name) {
+				valid_ip = false
```

Comment:
Given that we're checking if the interface is valid, I think a better name for this flag would be "valid_interface" or "valid_iface".

However, I think we can get rid of the flag completely with a little bit of restructuring.

e.g

```
if !regex.MatchString(iface.Name) {
  // This is a valid interface.  Get all the IPs from it.
  ips = append(ips, ipsForInterface(iface))
}
```